lci_strategic_plugin/src/lci_strategic_plugin.cpp

```diff
@@ -734,14 +746,22 @@ void LCIStrategicPlugin::planWhenWAITING(const cav_srvs::PlanManeuversRequest& r
 
   ROS_DEBUG("traffic_light_down_track %f", traffic_light_down_track);
   
+  auto bool_optional_late_certainty = canArriveAtGreenWithCertainty(current_state.stamp, traffic_light, true, false);
+  
+  if (!bool_optional_late_certainty)
+  {
+    throw std::invalid_argument("Unable to resolve the green signal...");
```

Comment:
Instead of throwing an exception just return another stop and wait maneuver to keep the vehicle in place while the light rolls over.